packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -110,10 +112,41 @@ class FlameChartState extends State<FlameChart> with AutoDisposeMixin {
 
     addAutoDisposeListener(controller.searchNotifier, () {
       setState(() {
-        final searchingValue = controller.search;
-        if (searchingValue.isNotEmpty) {
-          final Node node = findNode(searchingValue);
-          selectNode(node);
+        if (controller.search.isNotEmpty) {
+          if (controller.selectTheSearch) {
+            // Select the node.
+            final Node node = findNode(controller.search);
+            selectNode(node);
+
+            closeAutoCompleteOverlay();
+
+            controller.selectTheSearch = false;
+            controller.search = '';
+          } else {
+            // No exact match, return top matches.
+            var matches = matchNames(controller.search);
+            matches.sort();
+            matches = matches.sublist(0, min(topMatches, matches.length));
+
+            controller.searchAutoComplete.value = matches;
+          }
+        } else if (controller.selectTheSearch) {
+          // Escape hit, on empty search.
+          selectNode(null);
+          controller.selectTheSearch = false;
+        }
+      });
+    });
+
+    addAutoDisposeListener(controller.searchAutoCompleteNotifier, () {
+      setState(() {
+        if (autoCompleteOverlay == null) {
+          autoCompleteOverlay = createAutoCompleteOverlay(controller);
+          Overlay.of(context).insert(autoCompleteOverlay);
+        } else {
+          closeAutoCompleteOverlay();
+          autoCompleteOverlay = createAutoCompleteOverlay(controller);
+          Overlay.of(context).insert(autoCompleteOverlay);
```

Comment:
this code is duplicated in memory_heap_tree view. Consolidate somehow.

Reply:
Good catch.

```
addAutoDisposeListener(controller.searchAutoCompleteNotifier, () {
    setState(autoCompleteOverlaySetState(controller, context));
});


/// Helper setState callback when searchAutoCompleteNotifier changes, usage:
/// 
///     addAutoDisposeListener(controller.searchAutoCompleteNotifier, () {
///      setState(autoCompleteOverlaySetState(controller, context));
///     });
VoidCallback autoCompleteOverlaySetState(
  MemoryController controller,
  BuildContext context,
) {
  return () {
    if (autoCompleteOverlay != null) {
      closeAutoCompleteOverlay();
    }

    autoCompleteOverlay = createAutoCompleteOverlay(controller);
    Overlay.of(context).insert(autoCompleteOverlay);
  };
}
```